### PR TITLE
[pt] translation for comprehensions

### PR DESCRIPTION
**Pull request by ruan-brandao**
Related to [#355](https://github.com/doomspork/elixir-school/issues/355)

This lesson is not in the issue's list of pages to translate, but it doesn't have a translation.

Feel free to give me your feedback @pragmaticivan and anyone who wants to revise this translation.

### CODE REVIEW

**Comment by doomspork** [comment]
Thank you @ruan-brandao!  I'll give @pragmaticivan a chance to comment and then we can merge 👍

**Comment by pragmaticivan** [comment]
Ops, still reviewing.

**Comment by ruan-brandao** [comment]
BTW, I noticed that the lessons for Strings and Custom Mix Tasks need translation too. Should I translate them in this PR or open one PR per lesson?

**Comment by doomspork** [comment]
@ruan-brandao individual PRs are _usually_ easier for reviewers since they can be tackled in small chunks.

File: pt/lessons/basics/comprehensions.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+layout: page
+title: Comprehensions
+category: basics
+order: 13
+lang: pt
+---
+
+Comprehensions de listas são um 'syntactic sugar' (uma forma mais simples de escrever) para realizar loops em Enumerables em Elixir. Nessa lição veremos como podemos fazer iterações e gerar os resultados utilizando comprehensions.
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: What do you think about removing `de listas` in those cases?

Or just `Listas de Comprehensions`. ?

File: pt/lessons/basics/comprehensions.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+Em alguns casos comprehensions podem ser usadas para produzir código mais conciso para fazer iterações com `Enum` e `Stream`. Vamos começar olhando para uma comprehension simples e então observar suas várias partes:
+
+```elixir
+iex> list = [1, 2, 3, 4, 5]
+iex> for x <- list, do: x*x
+[1, 4, 9, 16, 25]
+```
+
+A primeira coisa que observamos é o uso de `for` e um generator (gerador). O que é um generator? Generators são as expressões `x <- [1, 2, 3, 4]` encontradas em comprehensions de listas. Eles são responsáveis por gerar o próximo valor.
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: Would be good to change here too, or remove `de listas` or add `listas de comprehensions`

File: pt/lessons/basics/comprehensions.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+3 - 1
+3 - 2
+3 - 3
+4 - 1
+4 - 2
+4 - 3
+4 - 4
+```
+
+Comprehensions de listas são _syntactic sugar_ e devem ser utilizadas apenas quando for apropriado.
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: Here too.

File: pt/lessons/basics/comprehensions.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+## Filtros
+
+Você pode pensar em filtros como um tipo de _guard_ para comprehensions. Quando um valor filtrado retorna `falso` ou `nil` ele é excluido da lista final. Vamos iterar por um intervalo e olhar apenas os números pares. Nós vamos usar a função `is_even/1` do módulo Integer para checar se um valor é par ou não.
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: use `false`, I think we should not translate data types.

File: pt/lessons/basics/comprehensions.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+E se nós quisermos produzir algo que não seja uma lista? Passando a opção `:into` nós podemos fazer exatamente isso! Como uma regra geral, `:into` aceita qualquer estrutura que implemente o protocolo `Collectable`.
+
+Usando `:into`, vamos criar um mapa de uma lista de palavras-chave.
+
+```elixir
+iex> for {k, v} <- [one: 1, two: 2, three: 3], into: %{}, do: {k, v}
+%{one: 1, three: 3, two: 2}
+```
+
+Como bitstrings implementam enumerable nós podemos usar comprehensions de listas e `:into` para criar strings:
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: Here too.

File: pt/lessons/basics/comprehensions.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+```elixir
+iex> for c <- [72, 101, 108, 108, 111], into: "", do: <<c>>
+"Hello"
+```
+
+É isso! Comprehensions de listas são um modo fácil de iterar por coleções de maneira concisa.
```
Comment (pragmaticivan) [inline pt/lessons/basics/comprehensions.md]: Here too.

File: pt/lessons/basics/comprehensions.md (hunk 1 above)
Comment (ruan-brandao) [inline pt/lessons/basics/comprehensions.md]: I liked removing `de listas`. I think it makes the text easier to read.

**Comment by pragmaticivan** [comment]
@doomspork ready!

**Comment by nscyclone** [comment]
Thanks, @ruan-brandao and @pragmaticivan!

**Pull request merged**